File: app/src/test/java/de/bahnhoefe/deutschlands/bahnhofsfotos/model/CountryTest.java
package de.bahnhoefe.deutschlands.bahnhofsfotos.model;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import java.util.Set;

class CountryTest {

    @Test
    void getCountryByCode() {
        var country = Country.getCountryByCode(Set.of(
                new Country("de", "Deutschland"),
                new Country("at", "Österreich")
        ), "at");

        assertThat(country).isNotEmpty();
        assertThat(country.get().getCode()).isEqualTo("at");
    }

    @ParameterizedTest
    @CsvSource(value = {
            "'', false",
            ", false",
            "'blah', true"
    })
    void hasTimetableUrlTemplate(String timetableUrlTemplate, boolean expectedValue) {
        var country = new Country("de", "Deutschland", null, timetableUrlTemplate);

        assertThat(country.hasTimetableUrlTemplate()).isEqualTo(expectedValue);
    }

}

File: app/src/test/kotlin/de/bahnhoefe/deutschlands/bahnhofsfotos/model/CountryTest.kt
package de.bahnhoefe.deutschlands.bahnhofsfotos.model

import de.bahnhoefe.deutschlands.bahnhofsfotos.model.Country.Companion.getCountryByCode
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

internal class CountryTest {
    @Test
    fun countryByCode() {
        val country = getCountryByCode(
            setOf(
                Country("de", "Deutschland"),
                Country("at", "Österreich")
            ), "at"
        )
        assertThat(country).isNotNull
        assertThat(country?.code).isEqualTo("at")
    }

    @ParameterizedTest
    @CsvSource(value = ["'', false", ", false", "'blah', true"])
    fun hasTimetableUrlTemplate(timetableUrlTemplate: String?, expectedValue: Boolean) {
        val country = Country("de", "Deutschland", null, timetableUrlTemplate)
        assertThat(country.hasTimetableUrlTemplate()).isEqualTo(expectedValue)
    }
}